packages/web-components/fast-components/src/styles/density.ts

```diff
@@ -0,0 +1,249 @@
+ * May be fractional.
+ *
+ * @defaultValue 2
+ *
+ * @public
+ */
+export const densityComponentHorizontalBetweenUnits = create<number>(
+    "density-component-horizontal-between-units"
+).withDefault(2);
+
+/**
+ * The design unit multiplier for adjustments to component horizontal density tokens, which adds the specified design units to the base setting.
+ *
+ * @remarks
+ * Ex: `1` to increase padding by one unit.
+ * May be fractional.
+ *
+ * @defaultValue 0
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnits = create<number>(
+    "density-component-horizontal-adjustment-units"
+).withDefault(0);
+
+/**
+ * The calculated cumulative value for adjustments to component horizontal density relative up the document hierarchy.
+ *
+ * @remarks
+ * To use relative density do not change this value, only consume it in styles.
+ * If you want to break the chain of relative density, override this default function with a fixed value.
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnitsCumulative = create<number>(
+    "density-component-horizontal-adjustment-units-cumulative"
+);
+densityComponentHorizontalAdjustmentUnitsCumulative.withDefault(
+    (element: HTMLElement) => {
+        const parent = composedParent(element);
+        const valParent = parent
+            ? densityComponentHorizontalAdjustmentUnitsCumulative.getValueFor(parent)
+            : 0;
+        const valThis = densityComponentHorizontalAdjustmentUnits.getValueFor(element);
+        return valParent + valThis;
+    }
+);
+
+/**
+ * The raw calculated measurement for outer padding not accounting for border thickness.
+ *
+ * @remarks
+ * Calculated from the base `densityComponentHorizontalOuterUnits`, the adjusted `densityComponentHorizontalAdjustmentUnits` and the `designUnit` size.
+ *
+ * @public
+ */
+export const densityComponentHorizontalOuterValue = create<number>(
+    "density-component-horizontal-outer-value"
+).withDefault(
+    (element: HTMLElement) =>
+        (densityComponentHorizontalOuterUnits.getValueFor(element) +
+            densityComponentHorizontalAdjustmentUnitsCumulative.getValueFor(element)) *
+        designUnit.getValueFor(element)
+);
+
+/**
+ * The calculated measurement for outer padding adjusted for border thickness of `strokeWidth`.
+ *
+ * @remarks
+ * Calculated from `densityComponentHorizontalOuterValue` - `strokeWidth`.
+ *
+ * @public
+ */
+export const densityComponentHorizontalOuterPadding = create<number>(
+    "density-component-horizontal-outer-padding"
+).withDefault(
+    (element: HTMLElement) =>
+        densityComponentHorizontalOuterValue.getValueFor(element) -
+        strokeWidth.getValueFor(element)
+);
+
+/**
+ * The inverse of the outer padding, used to pull child elements out into density padding.
+ *
+ * @remarks
+ * This is a convenience token to apply negative margins to a child element to pull it out over the container's padding.
```

Comment:
Can you help me understand this scenario? Negative margins are typically a code smell. What are we trying to accomplish with negative margin here?